pkg/linter/linter.go

```diff
@@ -263,9 +264,55 @@ func (l *Linter) Lint(ctx context.Context, m *manifest.Manifest) (Results, error
 		results = append(results, r...)
 	}
 
+	span.Debug("Getting versions for each mixin used in the manifest...")
+	err = l.validateVersionNumberConstraints(ctx, m)
+	if err != nil {
+		return nil, span.Error(err)
+	}
+
 	return results, nil
 }
 
+func (l *Linter) validateVersionNumberConstraints(ctx context.Context, m *manifest.Manifest) error {
+	for _, mixin := range m.Mixins {
+		if mapConfig, ok := mixin.Config.(map[string]interface{}); ok {
+			if v, exists := mapConfig["version"]; exists {
```

Comment:
Are there currently anything ensuring that mixins cannot use a config field called `version`?
I'm thinking of the situation where mixins already provide a field called `version`, which might result in unintended behaviour

Reply:
No, it was simply the suggested field name on the issue. We could always use `mixinVersion` or something more verbose to avoid clashing?

Reply:
I personally like `version` the best, but it could break some existing plugins.
@schristoff do you know of any way we can easily handle a "reservation" of a mixin configuration property?

Reply:
@kichristensen could a similar approach such as porter-state, porter-debug be taken? 
![image](https://github.com/getporter/porter/assets/26071258/0649f8b5-bcdf-45dc-9b05-57419a9fff04)